test/common/access_log/access_log_impl_test.cc

```diff
@@ -969,13 +969,27 @@ name: envoy.file_access_log
   path: /dev/null
   )EOF";
 
-  EXPECT_THROW_WITH_MESSAGE(
-      AccessLogFactory::fromProto(parseAccessLogFromV2Yaml(yaml), context_), EnvoyException,
-      "Unable to parse JSON as proto (INVALID_ARGUMENT:(filter.grpc_status_filter.statuses[0]): "
-      "invalid value \"NOT_A_VALID_CODE\" for type TYPE_ENUM): "
-      "{\"config\":{\"path\":\"/dev/"
-      "null\"},\"filter\":{\"grpc_status_filter\":{\"statuses\":[\"NOT_A_VALID_CODE\"]}},\"name\":"
-      "\"envoy.file_access_log\"}");
+  // The error message is serialized from JSON, and its fields can be ordered differently depending
+  // on where it is run. Thus we cannot use the standard EXPECT_THROW_WITH_MESSAGE, and instead
+  // check:
+  //   1. The statement throws
+  //   2. The exception is an EnvoyException
+  //   3. The EnvoyException contains a substring that uniquely identifies the kind of exception.
+  try {
```

Comment:
Is there a more canonical way to perform `EXPECT_THROW_WITH_MESSAGE`-like functionality on a substring, rather than comparing the whole string? This section feels cumbersome.

Reply:
we do have `EXPECT_THROW_WITH_REGEX` in `test/test_common/utility.h`

Reply:
Changed in 033d899.